app/src/main/java/swati4star/createpdf/fragment/PPTtoPdfFragment.java

```diff
@@ -0,0 +1,207 @@
+                .content(R.string.enter_file_name)
+                .input(getString(R.string.example), null, (dialog, input) -> {
+                    if (StringUtils.isEmpty(input)) {
+                        showSnackbar(mActivity, R.string.snackbar_name_not_blank);
+                    } else {
+                        final String inputName = input.toString();
+                        if (!mFileUtils.isFileExist(inputName + getString(R.string.pdf_ext))) {
+                            convertToPdf(inputName);
+                        } else {
+                            MaterialDialog.Builder builder = createOverwriteDialog(mActivity);
+                            builder.onPositive((dialog12, which) -> convertToPdf(inputName))
+                                    .onNegative((dialog1, which) -> openPPTtoPdf())
+                                    .show();
+                        }
+                    }
+                }).show();
+
+    }
+
+    @Override
+    public void onActivityResult(int requestCode, int resultCode, Intent data) {
+        mButtonClicked = false;
+        switch (requestCode) {
+            case mFileSelectCode:
+                if (resultCode == RESULT_OK) {
+                    mPPTFileUri = data.getData();
+                    mRealPath = RealPathUtil.getRealPath(getContext(), data.getData());
+                    showSnackbar(mActivity, getResources().getString(R.string.ppt_selected));
+                    String fileName = mFileUtils.getFileName(mPPTFileUri);
+                    if (fileName != null) {
+                        if (fileName.endsWith(Constants.pptExtension))
+                            mFileExtension = Constants.pptExtension;
+                        else if (fileName.endsWith(Constants.pptExtensionNew))
+                            mFileExtension = Constants.pptExtensionNew;
+                        else {
+                            showSnackbar(mActivity, R.string.extension_not_supported);
+                            return;
+                        }
+                    }
+                    fileName = getResources().getString(R.string.ppt_tv_view_text)
+                            + fileName;
+                    mTextView.setText(fileName);
+                    mTextView.setVisibility(View.VISIBLE);
+                    mCreatePdf.setEnabled(true);
+                    mMorphButtonUtility.morphToSquare(mCreatePdf,
+                            mMorphButtonUtility.integer());
+                }
+                break;
+        }
+        super.onActivityResult(requestCode, resultCode, data);
+    }
+
+    private void convertToPdf(String mFilename) {
+        String destinationPath = mSharedPreferences.getString(STORAGE_LOCATION,
+                getDefaultStorageLocation());
+        String mPath = destinationPath + mFilename
+                + mActivity.getString(R.string.pdf_ext);
+        try {
+            Presentation presentation = new Presentation(mRealPath);
```

Comment:
@Swati4star @vyankatesh24 The value of `mRealPath` is `null` here. I cannot understand why?

Reply:
Because the `RealpathUtil` class is unable to successfully convert `URI` to `filepath`.

Reply:
What can be done to fix this?

Reply:
Improve `getRealPath` function in `RealPathUtil` as per your need.

Reply:
But then in the Excel feature the path came out correctly. What is the reason for this time?